packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -140,6 +140,12 @@ class PubContext {
   }
 }
 
+enum PubOutputMode {
+  none,
+  standard,
```

Comment:
are these defined by pub? can you add a dartdoc to the enum, with briefly the difference between standard and summaryOnly?

Reply:
This was ad-hoc. `standard` would print everything from `pub get`, and `summaryOnly` would just set the `PUB_SUMMARY_ONLY` env var. I think I went with this enum at first because of some confusion regarding the previous behavior of `printProgress`, but https://github.com/flutter/flutter/pull/117896 smoothed that over.

I reverted this part of the change, and now just always set `PUB_SUMMARY_ONLY`, since I don't think there is any case where we want the full output of `pub get`.

Reply:
@nehalvpatel is this true, that we would never want the full output of `pub get`?

Reply:
I will defer to nehal for final approval, this general approach looks like the right one, my only request would be that we document in the code clearly what's happening.

Reply:
> @nehalvpatel is this true, that we would never want the full output of `pub get`?

Looking more carefully at all the references to `Pub.get`, I think I had myself confused and am now remembering why I chose the enum. Sorry.

Reply:
I think it might be desirable to omit `PUB_SUMMARY_ONLY` if a user is running `flutter pub <cmd>` from the CLI.

For now, the main concerning thing this silences is the enforce-lockfile hint, but I imagine it will lead to more divergences in the future. https://github.com/dart-lang/pub/blob/0ee46ee094d32e464528d5c31f8a7fbeab2821d8/lib/src/entrypoint.dart#L385

cc @sigurdm

Reply:
@andrewkolos I just saw your comment. Yea, I agree about using an enum or a different variable to avoid overloading `printProgress`. 🙂